contractsSol5/KyberNetwork.sol

```diff
@@ -317,46 +299,21 @@ contract KyberNetwork is Withdrawable2, Utils4, IKyberNetwork, ReentrancyGuard {
     function addKyberProxy(address networkProxy) external onlyAdmin {
         require(networkProxy != address(0), "proxy 0");
         require(!kyberProxyContracts[networkProxy], "proxy exists");
-        require(kyberProxyArray.length < MAX_APPROVED_PROXIES, "Max 2 proxy");
-
-        kyberProxyArray.push(networkProxy);
+        require(kyberStorage.addKyberProxy(networkProxy, MAX_APPROVED_PROXIES));
 
         kyberProxyContracts[networkProxy] = true;
+
         emit KyberProxyAdded(networkProxy);
     }
 
     function removeKyberProxy(address networkProxy) external onlyAdmin {
         require(kyberProxyContracts[networkProxy], "proxy not found");
 
-        uint proxyIndex = 2 ** 255;
+        require(kyberStorage.removeKyberProxy(networkProxy));
 
-        for (uint i = 0; i < kyberProxyArray.length; i++) {
-            if (kyberProxyArray[i] == networkProxy) {
-                proxyIndex = i;
-                break;
-            }
-        }
-
-        kyberProxyArray[proxyIndex] = kyberProxyArray[kyberProxyArray.length - 1];
-        kyberProxyArray.length--;
-
-        kyberProxyContracts[networkProxy] = false;
         emit KyberProxyRemoved(networkProxy);
     }
 
-    /// @notice should be called off chain
-    /// @dev get an array of all reserves
-    /// @return An array of all reserves
-    function getReserves() external view returns(IKyberReserve[] memory) {
-        return reserves;
-    }
-
-    /// @notice should be called off chain
-    /// @dev get an array of KyberNetworkProxies
-    /// @return An array of both KyberNetworkProxies
-    function getKyberProxies() external view returns(address[] memory proxies) {
```

Comment:
we still want these proxy array to be returned here.
so should call relevant function in kyberStorage.